src/api/routes/tx.ts

```diff
@@ -0,0 +1,80 @@
+import * as express from 'express';
+import { addAsync } from '@awaitjs/express';
+import * as Bluebird from 'bluebird';
+import { DataStore, DbTx } from '../../datastore/common';
+import { getTxFromDataStore } from '../controllers/db-controller';
+import { timeout, waiter } from '../../helpers';
+import * as cors from 'cors';
+
+export function createTxRouter(): express.Router {
+  const router = addAsync(express.Router());
+
+  router.use(cors());
+
+  router.getAsync('/', async (req, res) => {
+    const db: DataStore = req.app.get('db');
+    const txs = await db.getTxList();
+    const fullTxs = await Bluebird.mapSeries(txs.results, async tx => {
```

Comment:
Are there memory (or other) limitations that necessitate `mapSeries`? The `Bluebird.map` function supports a `concurrency` option.

Additionally, are there limitations that prevent us from getting all TX info (including events) via a single query with JOIN? Right now it looks like the number of queries is `n^2`

Reply:
After further reading, I see that we have a memory store and a PG store, which probably answers my questions.

Reply:
Yep -- the async map is a temp hack until the proper sql backend is implemented. This will use an optimized join queries when ready